File: Clases 2018/Clases-C#/ClasePractica11_2018 (update tabla Bugs - asignar bug)/C#_TPS_BugTracker/TPS_BugTracker/GUILayer/Bugs/frmAsignarBug.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using TPS_BugTracker.BusinessLayer.Entities;
using TPS_BugTracker.BusinessLayer.Services;
using TPS_BugTracker.GUILayer.Helper;

namespace TPS_BugTracker.GUILayer.Bugs
{
    public partial class frmAsignarBug : Form
    {
        private UsuarioService usuarioService = new UsuarioService();
     

        private Bug oBugSelected;
        private BugService oBugService;
        public frmAsignarBug()
        {
            InitializeComponent();

            oBugService = new BugService();

            GUIHelper.getHelper().llenarCombo(cbo_desa, usuarioService.consultarDesarrolladores(), "nombre", "id_usuario");
        }

        internal void mostrarBug(Bug bugSelected)
        {
            oBugSelected = bugSelected;
            txtIdBug.Text = oBugSelected.id_bug.ToString();
            txtTitulo.Text = oBugSelected.titulo;

        }

        private void btn_cancelar_Click(object sender, EventArgs e)
        {
            this.Close();
        }

        private void btn_aceptar_Click(object sender, EventArgs e)
        {   
            if (oBugSelected != null)            
            {
                HistorialBug historial = new HistorialBug();

                historial.asignado_a = (int) cbo_desa.SelectedValue;
                historial.fecha = DateTime.Now;
                historial.estado = 2;
                historial.responsable = oBugSelected.historial.Last().responsable;
                oBugSelected.addHistorial(historial);
                // Si tenemos un bug seleccionado significa que vamos a actualizar los cambios
                

                oBugService.actualizarBug(oBugSelected);
            }

            this.Close();
        }

        private void btn_cancelar_Click_1(object sender, EventArgs e)
        {
            this.Close();
        }
    }
}

File: Clases 2018/Clases-C#/Simulacro_parcial_2_2018 (creacion de bug e historico en transaccion)/VB_TPS_BugTracker/TPS_BugTracker/GUILayer/Bugs/frmAsignarBug.vb
Partial Public Class frmAsignarBug
    Inherits Form

    Private usuarioService As UsuarioService = New UsuarioService()
    Private oBugSelected As Bug
    Private oBugService As BugService

    Public Sub New()
        InitializeComponent()
        oBugService = New BugService()
        GUIHelper.getHelper().llenarCombo(cbo_desa, usuarioService.consultarDesarrolladores(), "nombre", "id_usuario")
    End Sub

    Friend Sub mostrarBug(ByVal bugSelected As Bug)
        oBugSelected = bugSelected
        txtIdBug.Text = oBugSelected.id_bug.ToString()
        txtTitulo.Text = oBugSelected.titulo
    End Sub

    Private Sub btn_cancelar_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btn_cancelar.Click
        Me.Close()
    End Sub

    Private Sub btn_aceptar_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btn_aceptar.Click
        If oBugSelected IsNot Nothing Then
            Dim historial As HistorialBug = New HistorialBug()
            historial.asignado_a = CInt(cbo_desa.SelectedValue)
            historial.fecha = DateTime.Now
            historial.estado = 2
            historial.responsable = oBugSelected.historial.Last().responsable
            oBugSelected.addHistorial(historial)
            oBugService.actualizarBug(oBugSelected)
        End If

        Me.Close()
    End Sub
End Class